API/Controllers/BeatmapsController.cs

```diff
@@ -1,35 +1,38 @@
 using API.DTOs;
-using API.Entities;
 using API.Services.Interfaces;
 using Asp.Versioning;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Http.HttpResults;
 using Microsoft.AspNetCore.Mvc;
 
 namespace API.Controllers;
 
 [ApiController]
 [ApiVersion(1)]
 [EnableCors]
-[Authorize(Roles = "system")] // Internal access only at this time
+[Authorize(Roles = "system, admin")] // Internal access only at this time
 [Route("api/v{version:apiVersion}/[controller]")]
 public class BeatmapsController(IBeatmapService beatmapService) : Controller
 {
     private readonly IBeatmapService _beatmapService = beatmapService;
 
-    [HttpGet("all")]
-    public async Task<ActionResult<IEnumerable<Beatmap>>> GetAllAsync() =>
-        Ok(await _beatmapService.GetAllAsync());
+    [HttpGet]
+    [EndpointSummary("List beatmaps")]
+    public async Task<Ok<IEnumerable<BeatmapDTO>>> ListAsync() =>
+        TypedResults.Ok(await _beatmapService.ListAsync());
 
-    [HttpGet("{beatmapId:long}")]
-    public async Task<ActionResult<Beatmap>> GetByOsuBeatmapIdAsync(long beatmapId)
+    [HttpGet("{key}")]
+    [EndpointSummary("Get a beatmap by versatile search")]
+    [EndpointDescription("Get a beatmap searching first by id, then by osu! beatmap id")]
+    public async Task<Results<NotFound, Ok<BeatmapDTO>>> GetAsync(string key)
```

Comment:
I feel like if the search is only being done by these two ids, we can use `int key` instead. Saves a parse call, and the database calls become simpler. e.g. if it's an int, send it to the database using `GetAsync`. If that fails, try to fetch by the long variant for osu id.